src/services/services.ts

```diff
@@ -3345,6 +3347,20 @@ export function createLanguageService(
         );
     }
 
+    function getImports(fileName: string): readonly string[] {
+        synchronizeHostData();
+        const file = getValidSourceFile(fileName);
+        const imports: Set<string> = new Set();

```

Comment:
let imports: Set<string> | undefined and allocated when needed?

Reply:
Efficiency isn't important in this code. It's called once every 3 minutes at most.